### PR TITLE
PLAT-121103: Add views in the qa-a11y sample

### PR DESCRIPTION
**Pull request by ybsung**
Add views in the qa-a11y sample

New components
- ArcPicker
- ArcSlider
- FanSpeedControl
- TemperatureControl
- WindDirectionControl
- Keypad
- MediaPlayer
- DatePicker
- TimePicker

Missing public components
- BodyText
- FullscreenPopup
- Heading
- IncrementSlider
- LabeledIconButton
- PopupMenu
- SliderButton
- TabbedPanels
- TabGroup
- ThumbnailItem
- ToggleButton

Additional consideration
- ~Add `disabled` props is TemperatureControl, WindDirectionControl, and FanSpeedControl samplers~
- Apply TabbpedPanels as the menu in the app

There are some bugs

Issue 1> PLAT-124516: [Agate] console error when tapping SliderButton
If there is no onChange in a SliderButton, there is the following error.

```
SliderButtonBehaviorDecorator.js:21 Uncaught TypeError: onChange is not a function
    at handleChange (SliderButtonBehaviorDecorator.js:21)
    at _class.forward (handle.js:391)
    at _class.<anonymous> (_curryN.js:37)
    at _class.<anonymous> (_arity.js:6)
    at _class.handle (handle.js:122)
```

Issue 2> PLAT-124521: [Agate] The text size in ArcSlider is bigger than the one in ArcPicker.

The slotCenter text is larger than the one that I expected.

Issue 3> PLAT-124524: [Agate] The `airup` icon position is different from `airdown` and `airright`

When selecting `airUp`, the icon moves

Issue 4> PLAT-124525: [Agate] Error when selecting `play` once and `next` several times

When selecting the play icon once and  the next icon several times, there is the following error.

```
Media.js:255 Uncaught (in promise) DOMException: The play() request was interrupted by a new load request. https://goo.gl/LdLk22
load @ Media.js:255
...
discreteUpdates @ react-dom.development.js:806
dispatchDiscreteEvent @ react-dom.development.js:4168
index.js:1 Uncaught Error: The error you provided does not contain a stack trace.
    at S (index.js:1)
    at V (index.js:1)
    at index.js:1
    at index.js:1
    at a (index.js:1)
```

Issue 5> PLAT-124526: [Agate] TemperatureControl is draggable even though it is disabled

Even though the TemperatureControl is disabled, the knob in the TemperatureControl could be draggable.

Issue 6> PLAT-124527: [Agate] Text is not aligned in center if no icon in FanSpeedControl

If there is no `icon` prop in the FanSpeedControl, the text should align in center.

Issue 7> PLAT-124529: [Agate] The title in PopupMenu is not distinguishable
I think it's out of scope but the title's color is black so it's not quite distinguishable.

Issue 8> PLAT-124530: [Agate] Error when selecting `5` in the Keypad
```
Uncaught TypeError: this.props.onChange is not a function
    at handleInputValue (Keypad.js:304)
    at onKeyButtonClick (Keypad.js:181)
    at forward (handle.js:391)
    at _curryN.js:37
    at _arity.js:6
    at adaptEvent (handle.js:696)
    at handle (handle.js:122)
    at Object.prepareHandleArgs [as onClick] (handle.js:213)
```

Enact-DCO-1.0-Signed-off-by: YB Sung (yb.sung@lge.com)

### CODE REVIEW

File: samples/qa-a11y/src/App/App.js [hunk 1]
```diff
@@ -46,45 +67,101 @@ const Menu = SpotlightContainerDecorator({enterTo: 'last-focused'}, 'div');
...
+
+		views.forEach((view, indexInViews) => {
+			const item = (
+				<Item
+					aria-label={view.title}
+					className={appCss.navItem}
+					data-menu={indexInViews}
+					key={indexInViews}
+					onClick={this.handleChangeView(indexInViews)}
+					slotBefore={('00' + indexInViews).slice(-2)}
```
Comment (MikyungKim) [inline samples/qa-a11y/src/App/App.js:150]: Could we have space between the number and the title?

File: samples/qa-a11y/src/views/ArcSlider.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+import Section from '../components/Section';
+
+import appCss from '../App/App.module.less';
+
+const ArcSliderView = () => (
+	<>
+		<Section title="Default">
+			<ArcSlider alt="Normal" />
+			<ArcSlider alt="With slotCenter" slotCenter="SlotCenter" />
```
Comment (MikyungKim) [inline samples/qa-a11y/src/views/ArcSlider.js:12]: SlotCenter looks overlapped with the slider. Would you please replace this with something else?

File: samples/qa-a11y/src/views/BodyText.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+			>
+				<BodyText>Text</BodyText>
+				<Button onClick={handleOpen(0, false)}>Close</Button>
+			</Popup>
+
+			<Popup
+				onClose={handleOpen(1, false)}
+				open={open[1]}
+			>
+				<BodyText disabled>Text</BodyText>
```
Comment (MikyungKim) [inline samples/qa-a11y/src/views/BodyText.js:31]: Does BodyText `disabled` prop? I don't think so...

File: samples/qa-a11y/src/views/DatePicker.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
+import DatePicker from '@enact/agate/DatePicker';
+import React from 'react';
+
+import Section from '../components/Section';
+
+const DatePickerView = () => (
+	<Section title="Default">
```
Comment (MikyungKim) [inline samples/qa-a11y/src/views/DatePicker.js:7]: No aria-labelled section in here?

File: samples/qa-a11y/src/views/FullscreenPopup.js [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+	const [open, handleOpen] = useBoolArray(5);
+
+	return (
+		<>
+			<Section title="Default">
+				<Button alt="Normal" onClick={handleOpen(0, true)}>Open 0</Button>
+				<Button alt="With noAnimation" onClick={handleOpen(1, true)}>Open 1</Button>
+				<Button alt="With 'clip' type" onClick={handleOpen(2, true)}>Open 2</Button>
+				<Button alt="With 'fade' type" onClick={handleOpen(3, true)}>Open 3</Button>
+				<Button alt="With 'slide' type" onClick={handleOpen(3, true)}>Open 4</Button>
```
Comment (MikyungKim) [inline samples/qa-a11y/src/views/FullscreenPopup.js:19]: handpeOpen(4, ?

File: samples/qa-a11y/src/views/FullscreenPopup.js [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+				<Button alt="Disabled" onClick={handleOpen(3, true)}>Open 5</Button>
```
Comment (MikyungKim) [inline samples/qa-a11y/src/views/FullscreenPopup.js:20]: handleOpen(5, ?

File: samples/qa-a11y/src/views/Heading.js [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+import React from 'react';
+
+import Section from '../components/Section';
+
+import appCss from '../App/App.module.less';
+
+const HeadingView = () => (
+	<>
+		<Section title="Default">
+			<Heading alt="Normal" size="large" showLine>Text 0</Heading>
```
Comment (MikyungKim) [inline samples/qa-a11y/src/views/Heading.js:11]: Does Heading itself could get a focus?
Do you need help from other component like you did in BodyText?

File: samples/qa-a11y/src/views/Keypad.js [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+
+const KeypadView = () => (
+	<>
+		<Section title="Default">
+			<Keypad alt="Normal" />
+			<Keypad alt="Disabled" disabled />
+		</Section>
+
+		<Section className={appCss.marginTop} title="Aria-labelled">
+			<Keypad alt="Aria-labelled" aria-label="This is a Label." />
```
Comment (MikyungKim) [inline samples/qa-a11y/src/views/Keypad.js:16]: I think we should think about how we provide aria-label for this component fist.
:duh:

Comment (MikyungKim) [reply samples/qa-a11y/src/views/Keypad.js:16]: I think it's out of this ticket scope but I saw
```Uncaught TypeError: this.props.onChange is not a function
    at handleInputValue (Keypad.js:304)
    at onKeyButtonClick (Keypad.js:181)
    at forward (handle.js:391)
    at _curryN.js:37
    at _arity.js:6
    at adaptEvent (handle.js:696)
    at handle (handle.js:122)
    at Object.prepareHandleArgs [as onClick] (handle.js:213)
```

File: samples/qa-a11y/src/views/MediaPlayer.js [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+
+const source = audioFiles.map((audioFile, index) => (<source key={index} src={audioFile} type="audio/mp3" />));
+
+const MediaPlayerView = () => (
+	<>
+		<Section title="Default">
+			<MediaPlayer alt="Normal">
+				{source}
+			</MediaPlayer>
+			<MediaPlayer alt="Disabled" disabled>
```
Comment (MikyungKim) [inline samples/qa-a11y/src/views/MediaPlayer.js:25]: I think `disabled` prop is not provided yet..

File: samples/qa-a11y/src/views/MediaPlayer.js [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+				{source}
+			</MediaPlayer>
+		</Section>
+
+		<Section className={appCss.marginTop} title="Aria-labelled">
```
Comment (MikyungKim) [inline samples/qa-a11y/src/views/MediaPlayer.js:30]: Do we need this section for MediaPlayer?

File: samples/qa-a11y/src/views/MediaPlayer.js [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+	'https://sampleswap.org/mp3/artist/19139/MarkNine_In-my-Place-160.mp3',
+	'https://sampleswap.org/mp3/artist/47067/DJ-Masque_Dont-Forget-To-Be-Yourself-160.mp3'
+];
+
+const source = audioFiles.map((audioFile, index) => (<source key={index} src={audioFile} type="audio/mp3" />));
+
+const MediaPlayerView = () => (
+	<>
+		<Section title="Default">
+			<MediaPlayer alt="Normal">
```
Comment (MikyungKim) [inline samples/qa-a11y/src/views/MediaPlayer.js:22]: MediaPlayer seems too right in the sample.

File: samples/qa-a11y/src/views/PopupMenu.js [hunk 12]
```diff
@@ -0,0 +1,86 @@
...
+import useBoolArray from '../components/useBoolArray';
+
+const PopupMenuView = () => {
+	const [open, handleOpen] = useBoolArray(5);
+
+	return (
+		<>
+			<Section title="Default">
+				<Button alt="Normal" onClick={handleOpen(0, true)}>Open 0</Button>
+				<Button alt="With a Title" onClick={handleOpen(1, true)}>Open 1</Button>
```
Comment (MikyungKim) [inline samples/qa-a11y/src/views/PopupMenu.js:16]: I think it's out of scope but the title's color is black so it's not quite distinguishable.

File: samples/qa-a11y/src/views/SliderButton.js [hunk 13]
```diff
@@ -0,0 +1,22 @@
...
+import React from 'react';
+
+import Section from '../components/Section';
+
+import appCss from '../App/App.module.less';
+
+const SliderButtonView = () => (
+	<>
+		<Section title="Default">
+			<SliderButton alt="Normal" onChange={Function.prototype}>{['A', 'B', 'C']}</SliderButton>
```
Comment (MikyungKim) [inline samples/qa-a11y/src/views/SliderButton.js:11]: Is `onChange` prop required? I saw your mention in the description but I think we should fix the error instead of this.

File: samples/qa-a11y/src/views/TabbedPanels.js [hunk 14]
```diff
@@ -0,0 +1,71 @@
...
+
+	const handleNoAnimation = () => setNoAnimation(!noAnimation);
+	const handleNoCloseButton = () => setNoCloseButton(!noCloseButton);
+	const handleVertical = () => setVertical(!vertical);
+
+	return (
+		<TabbedPanels
+			noAnimation={noAnimation}
+			noCloseButton={noCloseButton}
+			orientation={vertical ? 'vertical' : 'horizontal'}
```
Comment (MikyungKim) [inline samples/qa-a11y/src/views/TabbedPanels.js:28]: When `vertical`, it seems linked with the menu panels.

File: samples/qa-a11y/src/views/TemperatureControl.js [hunk 15]
```diff
@@ -0,0 +1,22 @@
...
+
+const TemperatureControlView = () => (
+	<>
+		<Section title="Default">
+			<TemperatureControl alt="Normal" />
+			<TemperatureControl alt="Disabled" disabled />
+		</Section>
+
+		<Section className={appCss.marginTop} title="Aria-labelled">
+			<TemperatureControl alt="Aria-labelled" aria-label="This is a Label." />
```
Comment (MikyungKim) [inline samples/qa-a11y/src/views/TemperatureControl.js:16]: Maybe update or add aria-label usage after implementation.

File: samples/qa-a11y/src/views/TimePicker.js [hunk 16]
```diff
@@ -0,0 +1,13 @@
+import DateTimePicker from '@enact/agate/DateTimePicker';
```
Comment (MikyungKim) [inline samples/qa-a11y/src/views/TimePicker.js:1]: I think we need a `TimePicker` instead of `DateTimePicker` here.

File: samples/sampler/stories/default/FanSpeedControl.js [hunk 17]
```diff
@@ -1,6 +1,6 @@
 import {mergeComponentMetadata} from '@enact/storybook-utils';
 import {action} from '@enact/storybook-utils/addons/actions';
-import {number, select} from '@enact/storybook-utils/addons/knobs';
+import {boolean, number, select} from '@enact/storybook-utils/addons/knobs';
```
Comment (MikyungKim) [inline samples/sampler/stories/default/FanSpeedControl.js:3]: Why did you change sampler in this PR?
Is this needed by your change?
Let's do this when we add `disabled` prop for these components.

**Review by MikyungKim — CHANGES_REQUESTED** [review]
Please find my comments. Thanks.

File: samples/qa-a11y/src/views/FullscreenPopup.js (hunk 5 above)
Comment (ybsung) [reply samples/qa-a11y/src/views/FullscreenPopup.js:19]: Fixed

File: samples/qa-a11y/src/views/FullscreenPopup.js (hunk 6 above)
Comment (ybsung) [reply samples/qa-a11y/src/views/FullscreenPopup.js:20]: Fixed

File: samples/qa-a11y/src/views/MediaPlayer.js (hunk 9 above)
Comment (ybsung) [reply samples/qa-a11y/src/views/MediaPlayer.js:25]: Removed.

File: samples/qa-a11y/src/views/PopupMenu.js (hunk 12 above)
Comment (ybsung) [reply samples/qa-a11y/src/views/PopupMenu.js:16]: I wrote it in the PR description.

File: samples/qa-a11y/src/views/TemperatureControl.js (hunk 15 above)
Comment (ybsung) [reply samples/qa-a11y/src/views/TemperatureControl.js:16]: Removed

File: samples/sampler/stories/default/FanSpeedControl.js (hunk 17 above)
Comment (ybsung) [reply samples/sampler/stories/default/FanSpeedControl.js:3]: Reverted.

File: samples/qa-a11y/src/views/TemperatureControl.js (hunk 15 above)
Comment (ybsung) [reply samples/qa-a11y/src/views/TemperatureControl.js]: Removed

File: samples/qa-a11y/src/views/SliderButton.js (hunk 13 above)
Comment (ybsung) [reply samples/qa-a11y/src/views/SliderButton.js:11]: onChange={Function.prototype} was removed.

File: samples/qa-a11y/src/views/MediaPlayer.js (hunk 10 above)
Comment (ybsung) [reply samples/qa-a11y/src/views/MediaPlayer.js:27]: Removed.

File: samples/qa-a11y/src/views/MediaPlayer.js (hunk 9 above)
Comment (ybsung) [reply samples/qa-a11y/src/views/MediaPlayer.js]: Removed.

File: samples/qa-a11y/src/views/Heading.js (hunk 7 above)
Comment (ybsung) [reply samples/qa-a11y/src/views/Heading.js:11]: Fixed like BodyText.

File: samples/qa-a11y/src/views/Keypad.js (hunk 8 above)
Comment (ybsung) [reply samples/qa-a11y/src/views/Keypad.js:16]: Removed the aria-labelled section first.

Comment (ybsung) [reply samples/qa-a11y/src/views/Keypad.js:16]: I wrote it in the PR description.

File: samples/qa-a11y/src/views/MediaPlayer.js (hunk 11 above)
Comment (ybsung) [reply samples/qa-a11y/src/views/MediaPlayer.js:22]: I hope we could go ahead.

File: samples/qa-a11y/src/views/TabbedPanels.js (hunk 14 above)
Comment (ybsung) [reply samples/qa-a11y/src/views/TabbedPanels.js:28]: It is the GUI issue in qa-a11y. I hope that we could resolve it in a separate issue if we want to fix it.

File: samples/qa-a11y/src/views/DatePicker.js (hunk 4 above)
Comment (ybsung) [reply samples/qa-a11y/src/views/DatePicker.js]:
```
<DatePicker alt="Aria-labelled" dayAriaLabel="day" monthAriaLabel="month" yearAriaLabel="year" />
```
Add aria-labelled section with custom aria label props.

File: samples/qa-a11y/src/App/App.js (hunk 1 above)
Comment (ybsung) [reply samples/qa-a11y/src/App/App.js]: Applied wider width for the slotBefore.

File: samples/qa-a11y/src/views/ArcSlider.js [hunk 18]
```diff
@@ -0,0 +1,26 @@
...
+
+import Section from '../components/Section';
+
+import appCss from '../App/App.module.less';
+
+const ArcSliderView = () => (
+	<>
+		<Section title="Default">
+			<ArcSlider alt="Normal" />
+			<ArcSlider alt="With slotCenter" slotCenter="Text" />
```
Comment (MikyungKim) [inline samples/qa-a11y/src/views/ArcSlider.js:12]: Let's make some space between ArcSliders to interact easily.

File: samples/qa-a11y/src/views/MediaPlayer.js (hunk 11 above)
Comment (MikyungKim) [reply samples/qa-a11y/src/views/MediaPlayer.js:20]: Let's remove alt..

File: samples/qa-a11y/src/views/TabbedPanels.js (hunk 14 above)
Comment (MikyungKim) [reply samples/qa-a11y/src/views/TabbedPanels.js:28]: Can we have a left margin from TabbedPanels? to distinguish from the menu selector.

**Review by MikyungKim — APPROVED** [review]
LGTM

File: samples/qa-a11y/src/views/SliderButton.js [hunk 19]
```diff
@@ -0,0 +1,22 @@
+import SliderButton from '@enact/agate/SliderButton';
+import React from 'react';
+
+import Section from '../components/Section';
+
+import appCss from '../App/App.module.less';
+
+const SliderButtonView = () => (
```
Comment (jeonghee27) [inline samples/qa-a11y/src/views/SliderButton.js:8]: I got console.error when i select button.

Uncaught TypeError: onChange is not a function
